diff -Naur org/configure.in new/configure.in
--- org/configure.in
+++ new/configure.in
@@ -1,7 +1,7 @@
 
 AC_REVISION($Revision: 1.177 $)
 
-AC_INIT([DSL CPE API driver],[4.21.1-pd3],[],[drv_dsl_cpe_api])
+AC_INIT([DSL CPE API driver],[4.21.1-pd4],[],[drv_dsl_cpe_api])
 
 AC_CONFIG_SRCDIR(src/Makefile.am)
 AM_INIT_AUTOMAKE
diff -Naur org/src/include/drv_dsl_cpe_autoboot.h new/src/include/drv_dsl_cpe_autoboot.h
--- org/src/include/drv_dsl_cpe_autoboot.h
+++ new/src/include/drv_dsl_cpe_autoboot.h
@@ -164,15 +164,15 @@ DSL_Error_t DSL_DRV_AutobootHandleDeviceEvt(
 #define DSL_AUTOBOOT_TRAINING_POLL_TIME 1000
 #define DSL_AUTOBOOT_INIT_POLL_TIME 100
 #define DSL_AUTOBOOT_SHOWTIME_POLL_TIME 1000
-#define DSL_AUTOBOOT_EXCEPTION_POLL_TIME 0
+#define DSL_AUTOBOOT_EXCEPTION_POLL_TIME 10
 #define DSL_AUTOBOOT_DIAGNOSTIC_POLL_TIME 1000
-#define DSL_AUTOBOOT_FW_REQUEST_POLL_TIME 0
+#define DSL_AUTOBOOT_FW_REQUEST_POLL_TIME 10
 #define DSL_AUTOBOOT_FW_WAIT_POLL_TIME 100
 #define DSL_AUTOBOOT_LINK_ACTIVATE_WAIT_POLL_TIME 100
 #define DSL_AUTOBOOT_CONFIG_WRITE_WAIT_POLL_TIME 100
 #define DSL_AUTOBOOT_RESTART_WAIT_POLL_TIME 100
 #define DSL_AUTOBOOT_FW_READY_POLL_TIME 100
-#define DSL_AUTOBOOT_RESTART_POLL_TIME 0
+#define DSL_AUTOBOOT_RESTART_POLL_TIME 100
 #define DSL_AUTOBOOT_ORDERLY_SHUTDOWN_POLL_TIME 250
 #define DSL_AUTOBOOT_DISABLE_POLL_TIME 100
 
